controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -44,12 +66,26 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
   virtual
   ~ControllerInterface() = default;
 
+  CONTROLLER_INTERFACE_PUBLIC
+  virtual
+  InterfaceConfiguration command_interface_configuration() const = 0;
+
+  CONTROLLER_INTERFACE_PUBLIC
+  virtual
+  InterfaceConfiguration state_interface_configuration() const = 0;
+
+  CONTROLLER_INTERFACE_PUBLIC
+  void assign_interfaces(
+    std::vector<hardware_interface::LoanedCommandInterface> && command_interfaces,
+    std::vector<hardware_interface::LoanedStateInterface> && state_interfaces);
+
+  CONTROLLER_INTERFACE_PUBLIC
+  void release_interfaces();
+
   CONTROLLER_INTERFACE_PUBLIC
   virtual
   return_type
-  init(
-    std::weak_ptr<hardware_interface::RobotHardware> robot_hardware,
-    const std::string & controller_name);
+  init(const std::string & controller_name);
```

Comment:
Reviewing the current controller portings in ros-control/ros2_controllers repository I realized, that this name argument is obsolete. Since controllers are `LifecycleNode`s they have access to their own name through the node name.

```suggestion
  init();
```

Reply:
#246 says "Hold my beer"